cli/src/program.rs

```diff
@@ -176,6 +178,13 @@ impl ProgramSubCommands for App<'_, '_> {
                                 .long("allow-excessive-deploy-account-balance")
                                 .takes_value(false)
                                 .help("Use the designated program id even if the account already holds a large balance of SOL")
+                        )
+                        .arg(
+                            Arg::with_name("enable_sol_alloc_free")
```

Comment:
@CriesofCarrots, @jstarry  By default after these changes are released, the cli will no longer deploy programs that depend on sol_alloc_free syscall unless the caller specifies this hidden flag.  This should be no issue for rust developers because rust programs haven't used this syscall for a long time.  C programs that explicitly call alloc/free will fail to deploy unless they pick up the latest SDK.  What do you think about this behavior, think we should instead default to allowing it for now?

Reply:
I agree with Justin, and also lean toward no flag at all.
If you do feel the hidden flag is necessary, can we add some documentation about when it will no longer work, and when it can be removed?

Reply:
The only reason we would need the flag is if someone is building with an older tool chain but deploying it with this version and doing so before the alloc_free disable feature gate is activated